src/Forms/EmailField.php

```diff
@@ -18,32 +21,26 @@ public function Type()
     }
 
     /**
-     * Validates for RFC 2822 compliant email addresses.
-     *
-     * @see http://www.regular-expressions.info/email.html
-     * @see http://www.ietf.org/rfc/rfc2822.txt
-     *
      * @param Validator $validator
      *
      * @return string
      */
     public function validate($validator)
     {
-        $result = true;
         $this->value = trim($this->value ?? '');
 
-        $pattern = '^[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$';
-
-        // Escape delimiter characters.
-        $safePattern = str_replace('/', '\\/', $pattern ?? '');
+        $result = true;
+        $message = _t('SilverStripe\\Forms\\EmailField.VALIDATION', 'Please enter an email address');
+        $validationResult = ConstraintValidator::validate(
+            $this->value,
+            new Constraints\Email(message: $message)
+        );
 
-        if ($this->value && !preg_match('/' . $safePattern . '/i', $this->value ?? '')) {
+        if (!$validationResult->isValid()) {
             $validator->validationError(
                 $this->name,
-                _t('SilverStripe\\Forms\\EmailField.VALIDATION', 'Please enter an email address'),
-                'validation'
+                $validationResult->getMessages()[0]['message'],
```

Comment:
Getting the message from the $validationResult rather than simply using $message in case we want to use the special symfony strings such as `{{ value }}` in the future